src/bootstrap/dist.rs

```diff
@@ -1058,7 +1058,11 @@ impl Step for RustAnalyzer {
         }
 
         let rust_analyzer = builder
-            .ensure(tool::RustAnalyzer { compiler, target, extra_features: Vec::new() })
+            .ensure(tool::RustAnalyzer {
+                compiler,
+                target,
+                extra_features: vec!["in-rust-tree".to_owned()]
```

Comment:
Rather then adding this at every call site, you could make RustAnalyzer its own tool outside of the macro, which calls ensure(ToolBuild) internally with the right features.

Reply:
I'm now working on the real PR and I'm trying to to understand this comment.

Do you mean adding a separate struct (what would it be named?) that implements `Step` manually? And the macro-generated `Step` would call it, like `Rls` ensures `Clippy`?

I'm trying to wrap my mind around this.

Reply:
Ohh do you mean not using the macro at all?

Reply:
correct, not using the macro at all. or you can change the macro not to have `extra_features` in some cases only; but that seems more complicated. up to you which you want to do.